lib/request.js

```diff
@@ -399,6 +400,7 @@ module.exports = {
   get protocol() {
     const proxy = this.app.proxy;
     if (this.socket.encrypted) return 'https';
+    if (this.get('X-AppEngine-HTTPS') === 'on') return 'https';
```

Comment:
It should after `proxy` detect. Otherwise any one can send a request with the fake `X-AppEngine-HTTPS: on` header.